pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -316,6 +316,26 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
         );
     }
 
+    /// @dev Hook is called when someone swaps through vault.
+    /// @param swapRequest Swap params.
+    /// @param currentBalanceTokenIn Vault balance of token that is swapped into the pool.
+    /// @param currentBalanceTokenOut Vault balance of token that is swapped out of the pool.
+    /// @return Amount of tokens that the user will receive in return for their token.
+    function onSwap(
+        SwapRequest memory swapRequest,
+        uint256 currentBalanceTokenIn,
+        uint256 currentBalanceTokenOut
+    ) public override returns (uint256) {
+        //cannot swap out uninitialized token
+        require(minBalances[swapRequest.tokenOut] == 0, "Uninitialized token");
+
+        if (minBalances[swapRequest.tokenIn] != 0) {
+            currentBalanceTokenIn = minBalances[swapRequest.tokenIn];
```

Comment:
What is the logic behind this line?

Reply:
This is necessary for the calculation of the exchange rate. The math that calculates the amount of tokens that a user receives in exchange for NEW_TOKEN expects a number of arguments, among others the current balance in the pool of NEW_TOKEN (check [here](https://github.com/PrimeDAO/balancer-v2-monorepo/blob/feature/53/uninitialized-token-behavior/pkg/pool-weighted/contracts/WeightedMath.sol#L70)). We are passing this fake amount that represents 1% of the pool value to calculate the price. This is my current understanding of how Index v1 works, but there is an [issue](https://github.com/PrimeDAO/balancer-v2-monorepo/issues/45) to double check the correctness of this assumption.